lib/private/DB/Adapter.php

```diff
@@ -28,11 +28,25 @@ public function __construct($conn) {
 	/**
 	 * @param string $table name
 	 *
-	 * @return int id of last insert statement
+	 * @return int id of last insert statement, 0 in case there was no INSERT before or it failed to get the ID
 	 * @throws Exception
 	 */
-	public function lastInsertId($table) {
-		return (int)$this->conn->realLastInsertId($table);
+	public function lastInsertId($table, bool $allowRetry = true): int {
+		$return = $this->conn->realLastInsertId($table);
```

Comment:
```suggestion
		$return = (int)$this->conn->realLastInsertId($table);
```
realLastInsertId does not return an int (it passes the call to dbal directly I think)